core/types/subtyping.cc

```diff
@@ -809,7 +817,11 @@ TypePtr Types::glb(Context ctx, const TypePtr &t1, const TypePtr &t2) {
         bool rtl = a1->klass == a2->klass || a1->klass.data(ctx)->derivesFrom(ctx, a2->klass);
         bool ltr = !rtl && a2->klass.data(ctx)->derivesFrom(ctx, a1->klass);
         if (!rtl && !ltr) {
-            return AndType::make_shared(t1, t2); // we can as well return nothing here?
+            if (a1->klass.data(ctx)->isClass() && a2->klass.data(ctx)->isClass()) {
+                return Types::bottom();
+            } else {
+                return AndType::make_shared(t1, t2); // we can as well return nothing here?
+            }
```

Comment:
What's the point of this change? Would the `AndType` not behave as bottom in the case where the `if` condition holds?

Reply:
That's correct, it wasn't behaving as bottom. Being explicit and returning bottom means that some other reductions are triggered.